### DISCUSSION TITLE
Add user-wide secrets for Actions

### DISCUSSION THREAD

**Discussion by hwittenborn** [community/community#4136] [Actions]
I currently use Drone CI, and the one thing that I really love about it is [the ability to use user-wide secrets](https://discourse.drone.io/t/organization-secrets-work-for-users-too/7507).

This is probably one of the sole reasons I'm avoiding Actions for my builds, as needing to add the same secret to multiple repositories quickly becomes tedious as the number of repositories starts to rise.

Any chance secrets could be allowed on a per-user level?

**Comment by rajbos** [comment]
You know you can set them at the [organization level](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-an-organization), right? Still want to have them for a user as well, or even a team level: that would help as well and saves you the hassle of using them on the organization level and then adding each repo that can use it to the allow list of the secret.

**Reply by hwittenborn** [reply]
Yep, I was wanting to use secrets on a user-wide level.

**Reply by leighmcculloch** [reply]
The organization level feature is great for organizations, but for repositories not associated with an organization, there is no way to specify a secrets that can be used across multiple repositories.

**Comment by leighmcculloch** [comment]
>This is probably one of the sole reasons I'm avoiding Actions for my builds

Much the same for me migrating Google Cloud Build builds to Actions. The prospect of having to manage private keys for Google Cloud across a dozen GitHub repos, rather than manage a single auth with Google Cloud, is pretty scary.

**Reply by hwittenborn** [reply]
Exactly! It's not even necessarily that it's tedious to initially add keys, but rather just much more so when I need to change their values, where it can get chaotic real quick.

**Comment by WeihanLi** [comment]
I used to use the Azure DevOps pipelines, it has a `variable group` feature, it would let me manage my secrets across different pipelines.
Currently, there's the organization secret, maybe we could have a user secret too?
Really hope for that, thanks.

**Comment by Badisi** [comment]
Now that npm is enforcing new token lifetime limits, this missing feature requires me to frequently manually update my npm token across all my GitHub projects, which is quite cumbersome..

**Comment by John-smith167** [comment]
Good point — [Drone](https://jinghongdrone.com/) CI’s user-wide secrets are definitely convenient for managing multiple repos. GitHub Actions doesn’t support that yet, though organization-level secrets or external secret managers can help replicate the setup until a similar feature is added.

**Reply by hwittenborn** [reply]
Spam 🙅‍♂️

**Comment by kashaf112** [comment]
I totally get your point about managing secrets across multiple repositories—it can definitely get tedious. Speaking of streamlining things, I recently came across a company like [Conglin anodized aluminum extrusions](https://clinalu.com/product-category/anodized-aluminum-extrusion/)
 that does something similar in the manufacturing space: they provide ready-to-use, precision-engineered products that save time and effort for engineers and fabricators. It’s interesting to see parallels between workflow optimization in software and efficiency in production.

**Reply by ghostinhershell** [reply]
Hi @kashaf112, we’re glad you’re here 😀

Thanks for sharing with the community. Unfortunately, we currently do not allow self-promotion or advertising on the Community Discussions. We want to make sure there is space for users to ask questions without overwhelming them with other conversations. Thank you for helping us maintain a productive and tidy community for all our members.
